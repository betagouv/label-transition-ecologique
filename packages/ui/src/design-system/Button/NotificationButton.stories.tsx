import {Meta, StoryObj} from '@storybook/react';

import {NotificationButton} from './NotificationButton';
import {useRef} from 'react';

const meta: Meta<typeof NotificationButton> = {
  component: NotificationButton,
  argTypes: {
    notificationVariant: {
      control: {type: 'select'},
    },
    variant: {
      control: {type: 'select'},
    },
    size: {
      control: {type: 'select'},
    },
    icon: {
      control: {type: 'text'},
    },
    iconPosition: {
      control: {type: 'select'},
    },
  },
  args: {
    icon: 'leaf-line',
    notificationValue: 2,
  },
};

export default meta;

type Story = StoryObj<typeof NotificationButton>;

/** Bouton par défaut, sans aucune props renseignée. */
export const Default: Story = {};

/** Bouton avec ref */
export const WithRef: Story = {
  render: args => {
    const buttonRef = useRef();
    const onButtonClick = () => console.log(buttonRef.current);
    return (
      <NotificationButton {...args} ref={buttonRef} onClick={onButtonClick} />
    );
  },
};
