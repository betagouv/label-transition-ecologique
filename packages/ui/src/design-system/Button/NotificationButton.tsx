import {Ref, forwardRef} from 'react';
import {ButtonProps} from './types';
import {Button} from './Button';
import {Notification, NotificationVariant} from '../Notification';

export const NotificationButton = forwardRef(
  (
    {
      notificationValue,
      notificationVariant,
      ...props
    }: {
      notificationValue: number;
      notificationVariant?: NotificationVariant;
    } & ButtonProps,
    ref?: Ref<HTMLButtonElement | HTMLAnchorElement>
  ) => {
    return (
      <div className="relative w-fit h-fit">
        <Button ref={ref} {...props} />
        <Notification
          size="xs"
          variant={notificationVariant}
          number={notificationValue}
          classname="h-6 w-6 !p-0 justify-center absolute top-0 right-0 translate-x-1/3 -translate-y-1/3"
        />
      </div>
    );
  }
);
