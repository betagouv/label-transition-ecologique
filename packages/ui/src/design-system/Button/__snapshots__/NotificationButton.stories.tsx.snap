// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`design-system/Button/NotificationButton Default smoke-test 1`] = `
<div class="relative w-fit h-fit">
  <button class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8 p-3.5">
    <span class="ri-leaf-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] fill-white group-hover:fill-white">
    </span>
  </button>
  <div class="text-primary fill-primary bg-primary-3 text-xs inline-flex items-center gap-1 p-2 rounded-full border-2 border-grey-1 shadow h-6 w-6 !p-0 justify-center absolute top-0 right-0 translate-x-1/3 -translate-y-1/3">
    <span class="font-extrabold leading-none">
      2
    </span>
  </div>
</div>
`;

exports[`design-system/Button/NotificationButton WithRef smoke-test 1`] = `
<div class="relative w-fit h-fit">
  <button class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8 p-3.5">
    <span class="ri-leaf-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] fill-white group-hover:fill-white">
    </span>
  </button>
  <div class="text-primary fill-primary bg-primary-3 text-xs inline-flex items-center gap-1 p-2 rounded-full border-2 border-grey-1 shadow h-6 w-6 !p-0 justify-center absolute top-0 right-0 translate-x-1/3 -translate-y-1/3">
    <span class="font-extrabold leading-none">
      2
    </span>
  </div>
</div>
`;
